src/settings/components/GeneralSettings.tsx

```diff
@@ -11,15 +11,18 @@ import {
   TextComponent,
   ToggleComponent,
 } from "./SettingBlocks";
+import ChainManager from "@/LLMProviders/chainManager";
 
 interface GeneralSettingsProps {
   getLangChainParams: () => LangChainParams;
   encryptionService: EncryptionService;
+  chainManager: ChainManager;
```

Comment:
Should we pass `ChatModelManager` instead of the entire `ChainManager` through all these components? Is `ChainManager` needed for other cases?